File: zxmap/src/main/java/com/lt/zxmap/view/MeaListener.java
package com.lt.zxmap.view;

import com.zxmap.zxmapsdk.geometry.LatLng;
import com.zxmap.zxmapsdk.style.layers.Layer;

import java.util.List;

/**
 * Created by Xiangb on 2019/9/27.
 * 功能：
 */
public interface MeaListener {

    void updateSource(boolean updateTotal);

    void onPointChange();

    void onMapReAdd();

    int getGeoType();

    List<LatLng> getLatlngs();

    List<List<LatLng>> getTotalList();

    int getEditLatlngIndex();

    void startTranslate();

    void stopTranslate();

    boolean isTranslate();

    MeaOptManager getOptManager();

    MeaLayerManager getLayerManager();

    void onLayerAdd(String name, Layer layer);

    void onLayerRemove(String name);
}

File: module_map/src/main/java/com/zx/module_map/view/measure/MeaListener.kt
package com.gt.module_map.view.measure

import com.esri.arcgisruntime.geometry.Point
import com.esri.arcgisruntime.mapping.view.MapView

interface MeaListener {

    fun getMapView(): MapView

    fun updateGraphic()

    fun onPointChange()

    fun onMapReAdd()

    fun getGeoType(): MeasureView.GeoType

    fun getPointList(): ArrayList<Point>

    fun startTranslate()

    fun stopTranslate()

    fun isTranslate(): Boolean
}